test/e2e_federation/authn.go

```diff
@@ -35,16 +35,20 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 	f := fedframework.NewDefaultFederatedFramework("federation-apiserver-authn")
 
 	var _ = Describe("Federation API server authentication [NoCluster]", func() {
+		var (
+			clusters fedframework.ClusterSlice
+		)
 		BeforeEach(func() {
 			fedframework.SkipUnlessFederated(f.ClientSet)
+			clusters = f.GetRegisteredClusters()
 		})
 
 		It("should accept cluster resources when the client has certificate authentication credentials", func() {
 			fcs, err := federationClientSetWithCert()
 			framework.ExpectNoError(err)
 
 			nsName := f.FederationNamespace.Name
-			svc := createServiceOrFail(fcs, nsName, FederatedServiceName)
+			svc := createServiceOrFail(fcs, nsName, FederatedServiceName, clusters)
```

Comment:
OK, i understand your predicament. sorry for that. its fine to pass clusters. better than the alternative